code-challenges.js

```diff
@@ -24,12 +49,63 @@ var middleLetters2 = "rhinoceros"
 // Expected output: “oc”
 
 
+const middle = (array) => {
+ let splitted = array.split("")
+//  even number of letters
+ if (splitted.length %2==0)
+// probably string interpolation is not the best way to do it but it works :)
+ return `${splitted[Math.floor((splitted.length / 2)-1)]}${splitted[Math.floor(splitted.length / 2)]}`
+// odd numbers of letters
+else if (splitted.length %2 !==0)
+return splitted[Math.floor(splitted.length / 2)]
+}
+
+console.log("Challenge 3.1:",middle(middleLetters1))
+console.log("Challenge 3.2:",middle(middleLetters2))
+
+// bonus variable to double check
+// var middleLetters3 = "abstract"
+// // Expected output: “tr”
+// console.log("Challenge 3.3:",middle(middleLetters3))
 
 
 // --------------------4) READ CAREFULLY: Create a CLASS to get the area of a sphere. Create THREE spheres with different radi as test cases. Area of a sphere =  4πr^2 (four * pi * radi squared)
 
 
 
+class Sphere {
+    constructor(radius) {
+      this.pi = 3.14159265359
+      this.radius = radius
+     }
+
+      calcArea =  () => { 
+        return 4*3.14159265359*Math.pow(this.radius, 2)
+    }   
+  }
+
+var sphere1 = new Sphere(this.radius=2)
+var sphere2 = new Sphere(this.radius=5)
+var sphere3 = new Sphere(this.radius=123)
+
```

Comment:
The class looks great. The variable `this.radius` is used inside the class, so doesn't need to be referenced in your console logs. `var sphere1 = new Sphere(2)` will get the job done.